File: solutions/javascript/hamming/1/hamming.js
//
// This is only a SKELETON file for the 'Hamming' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const compute = (strand1,strand2) => {
  if (strand1.length!==strand2.length){throw new Error('strands must be of equal length')}
  let distance=0
  for (let i=0;i<strand1.length;i++){distance+=Number(!(strand1[i]===strand2[i]))}
  return distance
};

File: solutions/go/hamming/1/hamming.go
package hamming

import (
        "errors"
       )

func Distance(a, b string) (int, error) {
    if len(a) != len(b) {
        return 0, errors.New("Strand lengths not equal")
    }
    var distance int = 0
	for i, _ := range a {
        if a[i] != b[i] {
            distance++
        }
    }
    return distance, nil
}
